.github/workflows/deploy.yml

```diff
@@ -53,7 +54,13 @@ jobs:
       - name: Install SAM CLI
         run: script/github-action-awscli-installation
       - name: Run Deployment Script
-        run: script/deploy-canary-demo
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: script/deploy-canary-demo 
+      - name: Run Deployment Script Previous Major Version
+        if: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
This is nice but does that mean I have to update the demo URLs in canary config for prev major version (from 2.x to 3.x) everytime we have a new major version?
It is not as nice but should we just hard code to prev-major version so we don't have to update config.

Reply:
That's how it would work right now, but we would have to update the configs anyways for their meta data anyways, so that it's not updating 2x metrics when we're on 3x, this also allows us the option to support more then just the last version, but I do see your point and I wouldn't say I disagree with just hard coding it. We could make it pretty easy to update though by just making it a single variable.

Reply:
I think for metrics it is already prev_version or something hard-coded.  There is no 2.x metrics :D

Reply:
Ah ok, then I'll make it prev-major-version if we ever want to update it to be version number won't be a hard change.